packages/path_provider/path_provider_foundation/darwin/Classes/PathProviderPlugin.swift

```diff
@@ -39,6 +39,11 @@ public class PathProviderPlugin: NSObject, FlutterPlugin, PathProviderApi {
   #endif
     return path
   }
+
+  // Returns the path for the container of the specified app group.
+  func getContainerPath(appGroupIdentifier: String) -> String? {
+      return FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: appGroupIdentifier)?.path
```

Comment:
On macOS it looks like we need to do extra work to see if the sandboxed app can create the directory:
> If you call the method with an invalid group identifier, namely one for which you do not have an entitlement, the method still returns a URL of the expected form, but the corresponding group directory does not actually exist, nor can your sandboxed app create it. Therefore be sure to test that you can successfully access the returned URL before using it.

https://developer.apple.com/documentation/foundation/filemanager/1412643-containerurl

Reply:
Yeah that's why I skipped the macOS part. App extensions haven't seemed to come up much for macOS Flutter developers, but we can always add it in later on if it becomes a priority